modules/apps/headless-apio/content-space/content-space-apio-impl/src/main/java/com/liferay/content/space/apio/internal/architect/resource/ContentSpaceCollectionResource.java

```diff
@@ -90,24 +95,39 @@ public String getName() {
 		).build();
 	}
 
+	private void _checkViewPermission(PermissionChecker permissionChecker)
```

Comment:
this method is really similar to https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/service/permission/GroupPermissionImpl.java#L70-L78, maybe you can use it. Notice that the exception is not exactly the same

Reply:
Used here: https://github.com/brianchandotcom/liferay-portal/pull/61169